lib/content_decoder.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Xcribe.ContentDecoder do
+  @moduledoc """
+  Used to decode requests content to an elixir collection (map, list, etc)
+
+  Currently only json is supported and the module `Xcribe.JSON` is used to
+  decode given value.
+  """
+
+  alias Xcribe.ContentDecoder.UnknownType
+  alias Xcribe.JSON
+
+  @json_format_regex ~r{application\/json|application\/vnd\..*json}
+
+  @doc """
+  Decode value by the given content_type.
+
+      iex> ContentDecoder.decode!("{\"key\":\"value\"}", "application/json")
+      %{"key" => "value"}
+
+  An UnknownType excption is raised when given content_type is unknown.
+  """
+  def decode!(value, content_type) do
+    content_type
+    |> define_format()
+    |> decode_for(value)
+  end
+
+  defp decode_for(:json, value), do: JSON.decode!(value)
+
+  defp define_format(content_type) do
+    if is_json?(content_type) do
```

Comment:
Maybe using pattern matching for `define_format` function will be better than using this `if else` here. What do you think?

Reply:
I need to run a regex in the content to determine the type, the initial idea was use `cond` as:
```elixir
cond do 
  is_json?() -> :json
  is_xml?() -> :xml
  true -> raise excption
end
```
As now just json was implemented the `Credo` warned about use cond with just one condition and suggest use `if` instead. I tried think a way to use pattern mathcing there but I don't figured out a good way.